### ISSUE TITLE
credential issues

### ISSUE THREAD

**Issue by akdor1154** [GitJournal/GitJournal#623]
Hey
Having issues authenticating with  1.83.10+3533 on Android 12.

With an ed25519 key generated in gitjournal or in ConnectBot, i get Invalid Credentials, and the key in github reports never used.

With an rsa 2048 key generated with ConnectBot, i get an error saying rsa/sha1 no longer allowed, but this time the key in github reports used.
 
Both times github key is set up as a deploy key, not account-wide.

1) ed25519 issue seems clear
2) if i understand right, i think the sha1 issue is a problem with the ssh client (git journal), not my key. I believe it is possible for an ssh client to use sha256/512 with an existing ssh-rsa key, and that git journal should do this, and if it did then the sha1 error above would be avoided without users needing to change keys.

Sorry this sounds terse, typing on phone :)
Cheers

**Comment by doingandlearning** [comment]
I'm having the same issues. I've tried regenerating the keys several times.

**Comment by aravindps** [comment]
I am also having the same problem in iOS. 
Is the a fix coming on?

**Comment by vHanda** [comment]
Sorry.

 I haven't had much time to work on GitJournal during the month of April. This month of May seems much better, so - yes - I'll definitely try to look into this!

**Comment by maxprehl** [comment]
Having this issue too, I made a larger comment on #85 . - https://github.com/GitJournal/GitJournal/issues/85#issuecomment-1144675922

**Comment by aravindps** [comment]
Can u please fix this issue?

**Comment by Jak2k** [comment]
This issue makes the all unusable for me. Please fix it.

**Comment by aravindps** [comment]
I have moved to working copy now! seems more reliable and works with any editor.

**Comment by Jak2k** [comment]
> I have moved to working copy now! seems more reliable and works with any editor.

But it isn't available for android.

**Comment by midick** [comment]
Is there any progress on this?

**Comment by vHanda** [comment]
Yes. Kind of. I've been working on moving away from libgit2 to go-git, which hopefully doesn't suffer from this problem.

But my main reason for switching is that managing the cross compilation of libgit2, is quite challenging, and with go-git, I can more easily hook in many of the customizations that I want. (end-to-end encryption for example)

You can expect a newer version of Gitjournal with this go-git backend this month.

**Comment by midick** [comment]
this would be wonderful

**Comment by midick** [comment]
Hi,
Has this been fixed?
Still doesn't work for me

**Comment by midick** [comment]
this became a bit pricey for an App that I can't use so I am back to running vimwiki in Termux :/

**Comment by vHanda** [comment]
Apologies. The version in master is ready I just have been testing it out
for way too long. I'll make a release today.

Please feel free to use the CI to get the latest APK if you want. But I'll
make a release today!

**Comment by ryanpcmcquen** [comment]
I am still hitting this issue on iOS.
